app/src/test/java/com/mstudio/superheroarch/presentation/CharacterDetailViewModelTest.kt

```diff
@@ -83,4 +86,24 @@ class CharacterDetailViewModelTest {
         viewModel.onCharacterReceived(RickAndMortyRepositoryInstruments.givenACharacterEntity().toCharacterData())
         verify(view).showNoInternetConnection()
     }
+
+    @Test
+    fun `given character detail screen, when user marks character as favorite, then the character is shown as favorite`() = runTest {
+        viewModel.onCharacterReceived(RickAndMortyRepositoryInstruments.givenACharacterEntity().toCharacterData())
+        `when`(setFavCharacterUseCase.updateCharacterFavoriteStatus(true, 1)).thenReturn(Unit)
+
+        viewModel.onFavoriteClicked()
+
+        verify(view).showCharacterAsNonFavorite()
```

Comment:
this test shouldn't work, you are checking both status without doing anything between both checks. it should be something like

showAsNonFav
onFavClicked
showAsFav

but now you are doing the click and then checking both status :/